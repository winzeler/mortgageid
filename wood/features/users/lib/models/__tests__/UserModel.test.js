const { UserModel } = require('#features/users/lib/models/UserModel');

describe('UserModel', () => {
  describe('hasFlag', () => {
    it('should test to see if user has flags', () => {
      const user = new UserModel({ flags: {
        aFlag: 'somevalue',
        specificFlag: 'specific_value',
      } });

      expect(user.hasFlag('aFlag')).toBe(true);
      expect(user.hasFlag('specificFlag', 'specific_value')).toBe(true);
      expect(user.hasFlag('specificFlag', 'other_value')).toBe(false);
    });

    // ---------------------------------------------------------------------------------------------

    it('should test for a secure flags', () => {
      const user = new UserModel({ secure_flags: {
        aFlag: 'somevalue',
        specificFlag: 'specific_value',
      } });

      expect(user.hasFlag('aFlag')).toBe(true);
      expect(user.hasFlag('specificFlag', 'specific_value')).toBe(true);
      expect(user.hasFlag('specificFlag', 'other_value')).toBe(false);
    });

    // ---------------------------------------------------------------------------------------------

    it('should not serialize secure flags', () => {
      const user = new UserModel({
        flags: { insecureFlag: 'insecure_value' },
        secure_flags: { secureFlag: 'secure_value' },
      });

      expect(user.toJSON()).toMatchSnapshot({
        created_at: expect.any(String),
        updated_at: expect.any(String),
      });
    });
  });
});
