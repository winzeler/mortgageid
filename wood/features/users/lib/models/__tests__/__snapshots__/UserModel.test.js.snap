// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserModel hasFlag should not serialize secure flags 1`] = `
Object {
  "account_type": undefined,
  "created_at": Any<String>,
  "email": undefined,
  "email_confirmed": undefined,
  "flags": Object {
    "insecureFlag": "insecure_value",
  },
  "id": undefined,
  "last_logged_in_at": undefined,
  "name": undefined,
  "teams": Array [],
  "updated_at": Any<String>,
}
`;
